notification-server/src/google_cloud.rs

```diff
@@ -357,17 +363,213 @@ mod tests {
             server.url(),
             "/v1/projects/fake_project_id/messages:send".to_string()
         );
-        let notification_information = NotificationInformation::new(
-            AccountAddress::from_str("4FmiTW2L2AccyR9VjzsnpWFSAcohXWf7Vf797i36y526mqiEcp").unwrap(),
-            "100".to_string(),
-            Preference::CCDTransaction,
+        let notification_information =
+            NotificationInformation::CCD(CCDTransactionNotificationInformation::new(
+                AccountAddress::from_str("4FmiTW2L2AccyR9VjzsnpWFSAcohXWf7Vf797i36y526mqiEcp")
+                    .unwrap(),
+                "100".to_string(),
+            ));
+        assert!(gc
+            .send_push_notification("valid_device_token", &notification_information)
+            .await
+            .is_ok());
+        mock.assert();
+    }
+
+    #[tokio::test]
+    async fn test_send_push_notification_cis2_without_metadata_url() {
+        let mut server = mockito::Server::new_async().await;
+        let mock_provider = MockTokenProvider {
+            token_response: Arc::new("mock_token".to_string()),
+            should_fail:    false,
+        };
+
+        let expected_body = json!({
+            "message": {
+                "token": "test_token",
+                "data": {
+                    "recipient": "3kBx2h5Y2veb4hZgAJWPrr8RyQESKm5TjzF3ti1QQ4VSYLwK1G",
+                    "amount": "200",
+                    "type": "cis2-tx",
+                    "token_id": "ffffff",
+                    "contract_index": "111",
+                    "contract_name": "init_contract"
```

Comment:
@Radiokot do you prefer me to leave metadata out in case it does not exist?

Reply:
Events for tokens without metadata should be skipped as they can't be handled properly in the walllet.

Reply:
So to be clear, this will happen given that I make a successful request to the node and no metadata exists under such contract. So everything happens gracefully.

Personally I would prefer that you ask whether metadata exists and if it does not exist then show a subset of the information. I find this flow rather important to support because everything happens validly.

Reply:
Ok, let's send the message even if there is no metadata, we'll handle it on the client side once when the design is ready for this case.

Reply:
Sure. Do you want me to leave out the field or represent the field as null?

I can see that in the case of metadata serialisation it per default provides it with the value of null. 

I prefer to just leave it out, but I also get the argument of consistency.

Reply:
You can omit it. From the client perspective there is no difference, but we'll save a few bytes at least :)